src/book/02-system/01-foundations/01-overview.mdx

```diff
@@ -22,3 +22,42 @@ After motions is hardware IO. The hardware will output the commands and input th
 All these modules are connected using NUClear which is a message passing software architecture. More can be found on NUClear [here](https://nuclear.readthedocs.io/en/latest/).
 
 ![Flow of the NUbots codebase](./images/software-overview.svg)
+
+## Roles
+
+Roles in the main codebase are the building blocks for creating a binary to run on a robot. A role file lists all the modules to be used in that binary. If a module is not included in the role file, that module will not run. All roles can be found in the [roles folder](https://github.com/NUbots/NUbots/tree/master/roles) in the first level of the repository and have the extension `.role`. For specifics about the NUClear roles system, see the [NUClear page](/system/foundations/nuclear#nuclear-roles).
+
+When adding modules to the list in a role, use the syntax `namespace::module_name`. For example, if you want to include the [`SensorFilter` module](https://github.com/NUbots/NUbots/tree/master/module/input/SensorFilter), then check the namespace in [`SensorFilter.hpp`](https://github.com/NUbots/NUbots/blob/master/module/input/SensorFilter/src/SensorFilter.hpp). The namespace is `namespace module::input`, so we have `module::input::SensorFilter`. However, all modules have `module::` so we drop this part of the namespace. The line in the role is therefore `input::SensorFilter`.
```

Comment:
This is assuming that at all times namespaces match the folder hierarchy.

Reply:
To clarify Cameron's point - the folder hierarchy is actually what you need to specify. The actual namespace the module is in doesn't matter. This is perhaps too subtle a point to bother explaining on the overview page though - personally I would skate over the details

Reply:
I would specify that its the folder hierarchy that needs to be specified. The first time someone has a module whose namespace differs from the folder hierarchy is going to be extremely confused because they are doing exactly what the docs said they should do.

Reply:
Ah, I always thought it was a namespace thing. One sure learns a lot writing documentation 😆

Reply:
Updated so it is correct.